### PR TITLE
(small fix) Ensure Theano finds libgpuarray folders automatically in Windows with conda package

### PR DESCRIPTION
**Pull request by notoraptor**
( this is related to windows documentation PR #5433 )

Theano uses `distutils.sysconfig.get_python_inc()` to get Python headers folder (and a similar function for libraries folder), and on Windows it returns the `include` folder located inside the environment folder (e.g. `C:\donnees\programmes\Anaconda3\envs\python2\include` (PS: same value is returned when calling ` distutils.sysconfig.get_python_inc(plat_specific=True)`)).

But conda installs the `libgpuarray` package under a folder named `Library` inside the environment folder (e.g. `C:\donnees\programmes\Anaconda3\envs\python2\Library\include\gpuarray`). It seems this `Library` folder is specific to Windows, so that it is impossible to get it directly from Python.

This fix ensures that the `Library` folder on Windows is used to find libgpuarray headers and static libraries automatically, so that user has nothing more to do when installing libgpuarray through conda.

Then the windows documentation will not need anything related to "configure libgpuarray" or something like that.

@abergeron 

gpuarray

### CODE REVIEW

File: theano/gof/cmodule.py [hunk 1]
```diff
@@ -1584,7 +1584,12 @@ def std_include_dirs():
     python_inc_dirs = ([py_inc] if py_inc == py_plat_spec_inc
                        else [py_inc, py_plat_spec_inc])
     gof_inc_dir = os.path.abspath(os.path.dirname(__file__))
-    return numpy_inc_dirs + python_inc_dirs + [gof_inc_dir]
+    other_dirs = []
+    if sys.platform == 'win32':
+        alt_inc_dir = os.path.abspath(os.path.normpath(sys.exec_prefix + '\Library\include'))
+        if os.path.exists(alt_inc_dir) and os.path.isdir(alt_inc_dir):
+            other_dirs.append(alt_inc_dir)
+    return numpy_inc_dirs + python_inc_dirs + [gof_inc_dir] + other_dirs
```
Comment (abergeron) [inline theano/gof/cmodule.py]: I would rather place this hack in GpuArrayType than here.  That way we won't invoke it for things that don't require the new backend.

File: theano/gof/cmodule.py [hunk 2]
```diff
@@ -1602,6 +1607,9 @@ def std_lib_dirs_and_libs():
         # Also add directory containing the Python library to the library
         # directories.
         python_lib_dirs = [os.path.join(os.path.dirname(python_inc), 'libs')]
+        alt_inc_dir = os.path.abspath(os.path.normpath(sys.exec_prefix + '\Library\lib'))
+        if os.path.exists(alt_inc_dir) and os.path.isdir(alt_inc_dir):
+            python_lib_dirs.append(alt_inc_dir)
```
Comment (abergeron) [inline theano/gof/cmodule.py]: same

**Comment by abergeron** [comment]
Apparently this also is a problem on other platforms so we probably should just always do it.

**Comment by notoraptor** [comment]
Updated.

@mgermain do you have the same problems with your machine ? Could you check where is libgpuarray headers in your environment ?

**Comment by mgermain** [comment]
~/.conda/envs/my_root/include/gpuarray

File: theano/gpuarray/type.py [hunk 3]
```diff
@@ -470,7 +471,17 @@ def c_headers(self):
                 '<gpuarray_api.h>']
 
     def c_header_dirs(self):
-        return [pygpu.get_include(), numpy.get_include()]
+        other_dirs = []
+        alt_inc_dir = os.path.abspath(os.path.normpath(sys.exec_prefix + '/Library/include'))
```
Comment (lamblin) [inline theano/gpuarray/type.py]: On Linux, headers seem to be in `sys.exec_prefix + '/include'` instead.
Same for `lib`, by the way. The whole `/Library` in the path does not seem to be there.

Comment (abergeron) [inline theano/gpuarray/type.py]: Then we might need something system-dependant after all ...

Comment (notoraptor) [inline theano/gpuarray/type.py]: @lamblin the `/Library` path seems to exist only with Conda distribution and only on Windows, so it is more a specific fix for Conda on Windows.

There are indeed `include` and `lib` paths on Linux, but I thought it was correctly included by Theano, maybe not after all ... I'll make some tests on Ubuntu tomorrow.

**Comment by notoraptor** [comment]
I confirm the problem for @mgermain on Ubuntu when using conda package `pygpu`.

`distutils.sysconfig.get_python_inc()` returns

`/home/notoraptor/Programmes/anaconda3/envs/testTheano2/include/python2.7`

, but gpuarray headers are in 

`/home/notoraptor/Programmes/anaconda3/envs/testTheano2/include/gpuarray/`.

So, on both Linux and Windows, the conda package `libgpuarray is installed in some directories that Theano can't find. And directories to check are not the same on both platforms.

I think I will just add another checking for Linux also. But then, I don't know if there is a similar problem on MacOS, for example !


NB: Jenkins and Travis failures are related to flake8 errors, I will fix it.

**Comment by abergeron** [comment]
I think this is normal for conda since Theano is the only package that actually builds things at runtime.  Normally the headers are used in conda-build.

**Comment by abergeron** [comment]
Apparently jenkins failed because the disk is full.

**Comment by abergeron** [comment]
jenkins test this please

**Pull request merged**